### ISSUE TITLE
Doesn't work under non-login shell

### ISSUE THREAD

**Issue by mdierolf** [awslabs/amazon-ecr-credential-helper#161]
It appears that the credential helper expects the environment variable HOME= to be set, otherwise logins hangs for 30 seconds and then fail, see this bug with Nomad:
https://github.com/hashicorp/nomad/issues/5527

SystemD does not set HOME=, as HOME is a variable used only in login shells, and is set by the [login](http://man7.org/linux/man-pages/man1/login.1.html) service

As such, the credential helper is not usable with any application launched from a non-login shell.

In my opinion, when HOME is not set, the credential helper should default to storing its cache under /var.

Also, when running under a non-login shell, AWS credentials should either be passed in using environment variables or stored in /etc/aws

**Comment by samuelkarp** [comment]
Hi @mdierolf!  Can you try setting the `AWS_ECR_CACHE_DIR` environment variable to your preferred location and see if that resolves your issue?

**Comment by gittur** [comment]
ohh my God, thankyou so much, I've been stuck with this since the past 4 hours. Just needed to set the HOME env var in my docker-compose systemd unit file. and now docker-compose is able to pull images from ECR instad of giving me the "no basic auth credentials found" error.

**Comment by abwiggins** [comment]
I've spent quite some time trying to figure this out. I am able to have the credential provider resolve the request to ECR when using a login-shell. But, when I try to automate this using systemd I am not able to authenticate. I set the HOME & AWS_ECR_CACHE_DIR variable in my systemd service file but still no luck. How can I get this to work with a non-login shell? Am I missing ENV vars? The error I receive is ---> `Error response from daemon: Get https://<account number>.dkr.ecr.<account region>.amazonaws.com/v2/<image>/manifests/<tag>: no basic auth credentials` 

I do not receive this error when logged in and running the script that way. 

OS: Ubuntu 16.04

Systemd Service File:
```[Unit]
Description = Start ECR download
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
Environment="HOME=/home/aw" "AWS_ECR_CACHE_DIR=/home/aw/.ecr"
WorkingDirectory=/home/aw
ExecStart=/home/aw/download_ecr_docker_image.sh
User=aw
Group=aw

[Install]
WantedBy=multi-user.target
